src/mgr/ActivePyModules.cc

```diff
@@ -875,8 +879,71 @@ PyObject* ActivePyModules::with_perf_counters(
   f.close_section();
   return f.get();
 }
+// Holds a list of label pairs for a counter, [(level, shallow), (pooltype, replicated)]
+typedef vector<pair<string_view,string_view>> perf_counter_label_pairs;
 
-PyObject* ActivePyModules::get_counter_python(
+PyObject* ActivePyModules::with_perf_counters(
+    std::function<void(PerfCounterInstance& counter_instance, PerfCounterType& counter_type, PyFormatter& f)> fct,
+    const std::string &svc_name,
+    const std::string &svc_id,
+    const std::string_view counter_name,
+    const std::string_view sub_counter_name,
+    const perf_counter_label_pairs &labels) const
+{
+  PyFormatter f;
+  /*
+    The resolved counter path, they are of the format
+    <counter_name>.<sub_counter_name> If the counter name has labels, then they
+    are segregated via NULL delimters.
+
+    Eg:
+      - labeled counter:
+        "osd_scrub_sh_repl^@level^@shallow^@pooltype^@replicated^@.successful_scrubs_elapsed"
+      - unlabeled counter: "osd.stat_bytes"
+  */
+  std::string resolved_path;
+  // NOTE: This should ideally use the resolved_path, but since the resolved path
+  // contains NULL delimiters and we cannot have an array section whose key has
+  // NULL characters, we will just use the counter_name to extract the value at
+  // the call site
+  Formatter::ArraySection perf_counter_value_section(f, counter_name);
```

Comment:
I'm pretty sure that `PyFormatter::open_array_section()` never uses the name that is passed into it because it's impossible to associate a name with an entry in a Python list.  This means that there should be no need to attempt to distinguish between unlabeled and labeled variants here.

Reply:
thanks for the confirmation! will remove the note. But for the reasoning of why we need a new variant of `with_perf_counter` please see here: https://github.com/ceph/ceph/pull/61997#discussion_r1992815786